### PR TITLE
[SLA] PIM-5638: fix error on filter grid method

### PR DESCRIPTION
**Pull request by solivier**
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | No
| Added Behats                      | Yes
| Changelog updated                 | Yes
| Review and 2 GTM                  | No

### CODE REVIEW

File: features/Context/DataGridContext.php [hunk 1]
```diff
@@ -745,6 +745,17 @@ public function iClickBackToGrid()
     }
 
     /**
+     * @Then /^I click on import profile$/
+     */
+    public function iClickOnImportProfile()
+    {
+        $this->wait();
```
Comment (juliensnz) [inline features/Context/DataGridContext.php]: you can spin on... Oh well, 1.3 ^^

Comment (juliensnz) [inline features/Context/DataGridContext.php]: please remember to rework it on 1.4 :)

File: features/datagrid/products_category_backtothegrid.feature [hunk 2]
```diff
@@ -44,3 +44,9 @@ Feature: Product category back to the grid
   Scenario: Successfully display the no results found message
     Given I filter by "SKU" with value "novalues"
     Then I should see "No results found. Try to change your search criteria."
+
+  Scenario: Successfully apply category's filter on product grid without affecting other grids
+    Given I filter by "category" with value "winter_collection"
+    And I click on import profile
+    When I refresh the grid
+    Then I should not see "Server error"
```
Comment (juliensnz) [inline features/datagrid/products_category_backtothegrid.feature]: "I should not see the text" in 1.4

**Comment by juliensnz** [comment]
GTM

File: features/Context/DataGridContext.php (hunk 1 above)
Comment (solivier) [inline features/Context/DataGridContext.php]: yep 1.3 :(

File: features/datagrid/products_category_backtothegrid.feature (hunk 2 above)
Comment (solivier) [inline features/datagrid/products_category_backtothegrid.feature]: thanks for the reminder

File: src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php [hunk 3]
```diff
@@ -198,9 +198,10 @@ protected function getFiltersToApply(DatagridConfiguration $config)
             $filters[] = $this->getFilterObject($column, $filter);
         }
 
-        if (!isset($filtersConfig['category'])) {
+        if (!isset($filtersConfig['category']) && $config->getName() === 'product-grid') {
```
Comment (skeleton) [inline src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php]: yoda

**Comment by skeleton** [comment]
gtm. be really carefull when you'll merge it on 1.4

File: src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php [hunk 4]
```diff
@@ -198,18 +198,37 @@ protected function getFiltersToApply(DatagridConfiguration $config)
...
         }
 
         return $filters;
     }
 
     /**
+     * Return the category filter config for the given $gridname,
+     * if this $gridname is not filterable by category, return null.
+     *
+     * @param $gridname
```
Comment (skeleton) [inline src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php]: missing type

Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php]: string

**Comment by pierallard** [comment]
GTM

File: src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php [hunk 5]
```diff
@@ -198,18 +198,37 @@ protected function getFiltersToApply(DatagridConfiguration $config)
...
+     *
+     * @return array|null
+     */
+    protected function getCategoryFilterConfig($gridname)
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php]: Why change the case compared to l. 201 ?

File: src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php (hunk 4 above)
Comment (willy-ahva) [inline src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php]: string

File: src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php (hunk 5 above)
Comment (solivier) [inline src/Pim/Bundle/DataGridBundle/Extension/Filter/FilterExtension.php]: no reason i can change

**Pull request merged**